qbraid_qir/qasm3/visitor.py

```diff
@@ -432,7 +439,7 @@ def _visit_gate_definition(self, definition: QuantumGateDefinition) -> None:
             raise Qasm3ConversionError(f"Duplicate gate definition for {gate_name}")
         self._custom_gates[gate_name] = definition
 
-    def _visit_basic_gate_operation(self, operation: QuantumGate) -> None:
+    def _visit_basic_gate_operation(self, operation: QuantumGate, inverse: bool = False) -> None:
         """Visit a gate operation element.
```

Comment:
Can you add one sentence to the docstring explaining what the difference is between invoking this function with `inverse = True` vs. `inverse = False`? If `inverse = True` is only applicable to (or only has an "effect" on) a certain subset of operations, then we should either emit a warning or raise a `ValueError` when used improperly. Or split into separate functions and abstract shared functionality into a helper. Whichever is cleaner.

Reply:
I agree that the inverse only has "effect" on the rotation, S and T gates. 
I have handled this in the `map_qasm_inv_op_to_pyqir_callable` which returns the inverse operation given the gate name and what to do in this inversion. 

I think minimal changes are introduced when `inverse` is introduced as a param, so maybe we can go with the same implementation?